pkg/assembler/grammar.go

```diff
@@ -126,7 +131,15 @@ func (e *Expression) AsMathOperation() *MathOperation {
 }
 
 func (e *Expression) AsImmediate() *string {
-	return e.Immediate
+	if e.Immediate == nil {
+		return nil
+	}
+	if e.Immediate.Sign != "" {
+		immediate := signedString(e.Immediate.Sign == "-", *e.Immediate.Value)
+		return &immediate
+	}
+	immediate := signedString(false, *e.Immediate.Value)
+	return &immediate
```

Comment:
This all can be refactored into:
```go
imm := signedString(e.Immediate.Sign == "-", e.Immediate.Value)
return imm;
```

After this refactor there is no need for a `signedString` function as well. You can just put in `AsImmediate` instead:

```go
if e.Immediate.Sign == "-" {
   return the "-%s"
}
return the immediate
```